File: bannerlayout/src/main/java/com/bannerlayout/animation/StackTransformer.java
package com.bannerlayout.animation;

import android.view.View;

@SuppressWarnings("ALL")
public class StackTransformer extends ABaseTransformer {

    @Override
    protected void onTransform(View view, float position) {
        view.setTranslationX(position < 0 ? 0f : -view.getWidth() * position);
    }

}

File: banner-transformer/src/main/java/androidx/banner/transformer/internal/StackTransformer.kt
package androidx.banner.transformer.internal

import android.view.View
import androidx.banner.transformer.ABaseTransformer

class StackTransformer : ABaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        page.translationX = if (position < 0) 0f else -page.width * position
    }
}
